scanner/src/lib/dataproducts/LogisticsEmissions_v0_1.svelte

```diff
@@ -50,9 +50,106 @@
+    totalEmissionsIntensityRoadFreight / data.roadFreightEmissions.length
+  const averageEmissionsIntensitySeaFreight =
+    totalEmissionsIntensitySeaFreight / data.seaFreightEmissions.length
+  function averageEmissionsIntensity() {
+    if (isNaN(averageEmissionsIntensityRoadFreight)) {
+      if (isNaN(averageEmissionsIntensitySeaFreight)) {
+        return 0
+      } else {
+        return averageEmissionsIntensitySeaFreight
+      }
+    } else {
+      if (isNaN(averageEmissionsIntensitySeaFreight)) {
+        return averageEmissionsIntensityRoadFreight
+      } else {
+        return (averageEmissionsIntensityRoadFreight + averageEmissionsIntensitySeaFreight) / 2
+      }
+    }
+  }
+
+  const totalEmissionsPerTceSeaFreight = data.seaFreightEmissions.reduce(
+    (allSeaTotal, seaFreightEmission) => {
+      const seaTotal = seaFreightEmission.emissionsPerTce.reduce(
+        (acc, emissionPerTce) => acc + emissionPerTce.emissions,
+        0
+      )
+      return allSeaTotal + seaTotal
+    },
+    0
+  )
+  const totalEmissionsPerTceRoadFreight = data.roadFreightEmissions.reduce(
+    (allRoadTotal, roadFreightEmission) => {
+      const roadTotal = roadFreightEmission.emissionsPerTce.reduce(
+        (acc, emissionPerTce) => acc + emissionPerTce.emissions,
+        0
+      )
+      return allRoadTotal + roadTotal
+    },
+    0
+  )
+  const totalEmissionsPerTce = totalEmissionsPerTceRoadFreight + totalEmissionsPerTceSeaFreight
+
+  const roadEmissionSources = data.roadFreightEmissions.map((roadFreightEmission) =>
+    roadFreightEmission.emissionsPerTce.map((emissionPerTce) => emissionPerTce.source)
+  )
+  const seaEmissionSources = data.seaFreightEmissions.map((seaFreightEmission) =>
+    seaFreightEmission.emissionsPerTce.map((emissionPerTce) => emissionPerTce.source)
+  )
+  const emissionSources = removeDuplicates(roadEmissionSources.concat(seaEmissionSources).flat())
+  function removeDuplicates(array: Array<string>) {
+    return array.filter((value, index) => array.indexOf(value) === index)
+  }
 </script>
 
 <article>
+  {#if data.roadFreightEmissions.length > 0 || data.seaFreightEmissions.length > 0}
+    <div class="title">Total emissions of the transports</div>
+    <Road
+      origin={data.roadFreightEmissions[0].origin}
+      destination={data.roadFreightEmissions[data.roadFreightEmissions.length - 1].destination}
+      isTotal
+    />
+    <DataRow label="Total emissions" value={formatNumber(totalEmissions, "CO2e tonnes")} />
+    <DataRow
+      label="Total Emissions intensity"
+      value={formatNumber(totalEmissionsIntensity, "CO2e grams / tonne / km")}
+    />
+    <DataRow
+      label="Average Emissions intensity"
+      value={formatNumber(averageEmissionsIntensity(), "CO2e grams / tonne / km")}
+    />
+    <DataRow
+      label="Total Emissions Per TCE"
+      value={formatNumber(totalEmissionsPerTce, "CO2e tonnes")}
+    />
+    <DataRow label="Emissions sources" value={emissionSources} />
+  {/if}
```

Comment:
Is there a reason why we would ever want to not have total? Is this indicated somehow in the designs?

Reply:
It's not in the design but the existing implementation checks for the presence of road and sea freight emissions before rendering both so I am taking that precaution here too